### PR TITLE
virt-test: guest-hw.cfg: add variants for nettype.

### PR DESCRIPTION
**Pull request by CongSmile**
Signed-off-by: Cong Li <coli@redhat.com>

### CODE REVIEW

**Comment by CongSmile** [comment]
id: 1223195

File: shared/cfg/guest-hw.cfg [hunk 1]
```diff
@@ -298,3 +298,9 @@ variants:
             Windows:
                 use_ovmf_autounattend = yes
                 send_key_at_install = ret
+
+# Nettype
+variants:
+    - @bridge:
```
Comment (xutian) [inline shared/cfg/guest-hw.cfg]: we need  nettype = bridge here

**Comment by CongSmile** [comment]
For readability and maintainability, needs specify the nettype.

Hi @xutian 

Thanks for your explanation.

I have updated code, could you help review it again?

Thanks.

**Comment by xutian** [comment]
LGTM,
Acked-by: Xu Tian <xutian@redhat.com>

**Comment by ypu** [comment]
There are more nettype in the framework. Please add them all.

**Comment by CongSmile** [comment]
Hi @ypu 

I have already added them according to the definition.
Have I misunderstood about sth?

Could you help explain it?

Thanks.

        def add_net(devices, vlan, nettype, ifname=None, tftp=None,
                    bootfile=None, hostfwd=[], netdev_id=None,
                    netdev_extra_params=None, tapfds=None, script=None,
                    downscript=None, vhost=None, queues=None, vhostfds=None,
                    add_queues=None, helper=None, add_tapfd=None,
                    add_vhostfd=None, vhostforce=None):
            mode = None
            if nettype in ['bridge', 'network', 'macvtap']:
                mode = 'tap'
            elif nettype == 'user':
                mode = 'user'
            else:
                logging.warning("Unknown/unsupported nettype %s" % nettype)
                return ''

**Comment by xiagao** [comment]
Acked-by: Xiaoling Gao <xiagao@redhat.com>

**Comment by ypu** [comment]
@CongSmile my first comment is based on your first version which only have macvtap and bridge. Not based on your new patch.

**Comment by ypu** [comment]
When you add variants in guest-hw.cfg please remember to add the default value in tests.cfg files for all backends. Otherwise it will cause problem when we use ./run to run a test.

**Comment by CongSmile** [comment]
Hi @ypu 

Thanks for your review.

I have updated the code, could you help review it again?

Thanks.

**Comment by ypu** [comment]
Hi @CongSmile there are more than one tests.cfg in virt-test. Please update them all.

**Comment by CongSmile** [comment]
Hi @ypu 

Thanks for your review.
And I have updated all tests.cfg, could you help review it again?

Thanks.

**Comment by suqinhuang** [comment]
Acked-by: Suqin Huang

**Comment by spcui** [comment]
LGTM
Acked-by: @spcui

**Comment by xutian** [comment]
LGTM,

Acked-by: Xu Tian <xutian@redhat.com>

**Pull request merged**